exporter/exporterqueue/queue.go

```diff
@@ -0,0 +1,69 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package exporterqueue // import "go.opentelemetry.io/collector/exporter/exporterqueue"
+
+import (
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/exporter/internal"
+)
+
+// Queue defines a producer-consumer exchange which can be backed by e.g. the memory-based ring buffer queue
+// (boundedMemoryQueue) or via a disk-based queue (persistentQueue)
+// This API is at the early stage of development and may change without backward compatibility
+// until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
+type Queue[T any] internal.Queue[T]
+
+// Settings defines settings for creating a queue.
+type Settings[T any] struct {
+	Sizer            internal.Sizer[T]
+	Capacity         int
+	DataType         component.DataType
+	ExporterSettings exporter.CreateSettings
+}
+
+type Factory[T any] func(Settings[T]) Queue[T]
+
+// NewMemoryQueueFactory returns a factory to create a new memory queue.
+// This API is at the early stage of development and may change without backward compatibility
+// until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
+func NewMemoryQueueFactory[T any]() Factory[T] {
+	return func(set Settings[T]) Queue[T] {
+		return internal.NewBoundedMemoryQueue[T](internal.MemoryQueueSettings[T]{
+			Sizer:    set.Sizer,
+			Capacity: set.Capacity,
+		})
+	}
+}
+
+// PersistentQueueSettings defines developer settings for the persistent queue factory.
+// This API is at the early stage of development and may change without backward compatibility
+// until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
+type PersistentQueueSettings[T any] struct {
+	// Marshaler is used to serialize queue elements before storing them in the persistent storage.
+	Marshaler func(req T) ([]byte, error)
```

Comment:
Extract "Marshaler" as a type and eventually deprecate `RequestMarshaler` in the exporterhelper?

Reply:
Defined Marshaler in this package and deprecated `RequestMarshaler`